test/system/components/student/concept_map_test.rb

```diff
@@ -0,0 +1,59 @@
+require "application_system_test_case"
+
+module Components
+  module Student
+    class ConceptMapTest < ApplicationSystemTestCase
+      def setup
+        super
+
+        # This component uses the API, which requires authentication.
+        sign_in!
+      end
+
+      test "renders concept map for joined track" do
+        # TODO: Unskip once this issue is resolved:
+        # https://github.com/exercism/v3-website/issues/143
+        skip
+
+        track = create :track, title: "Ruby"
+        create :user_track, track: track, user: @current_user
+        basics, booleans, atoms = setup_concepts(track, 'basics', 'booleans', 'atoms')
+        lasagna, pacman, logger = setup_concept_exercises(track, 'lasagna', 'pacman', 'logger')
+
+        lasagna.taught_concepts << basics
+        pacman.taught_concepts << booleans
+        pacman.prerequisites << basics
+        logger.taught_concepts << atoms
+        logger.prerequisites << booleans
+
+        visit tracks_url
```

Comment:
So...

This test (file) is designed to use the `test_` url (e.g. https://github.com/exercism/v3-website/blob/master/test/system/components/student/tracks_list_test.rb#L18). Effectively (most of?) the fixture data is already written in that controller. We are then testing the component in isolation from the rest of everything else.

The test you've written here would fit very nicely to check that the flow works correctly in terms of things actually getting unlocked and displaying as unlocked (this could live in `test/system/flows/track_concepts.rb` or similar).

So maybe you want to copy this to that, and then strip this version down and remove the link-walking and fixtures, and rely on the test controller data?